File: leetcode/go/_155/main.go
package main

import "fmt"

func main() {
	minStack := Constructor()
	minStack.Push(-2)
	minStack.Push(0)
	minStack.Push(-3)
	fmt.Println(minStack.GetMin())
	minStack.Pop()
	fmt.Println(minStack.Top())
	fmt.Println(minStack.GetMin())
}

type MinStack struct {
	stack    []int
	minStack []int
}

func Constructor() MinStack {
	return MinStack{stack: []int{}, minStack: []int{}}
}

func (m *MinStack) Push(val int) {
	m.stack = append(m.stack, val)

	if len(m.minStack) == 0 {
		m.minStack = append(m.minStack, val)
	} else {
		m.minStack = append(m.minStack, min(m.minStack[len(m.minStack)-1], val))
	}
}

func (m *MinStack) Pop() {
	m.stack = m.stack[:len(m.stack)-1]
	m.minStack = m.minStack[:len(m.minStack)-1]
}

func (m *MinStack) Top() int {
	return m.stack[len(m.stack)-1]
}

func (m *MinStack) GetMin() int {
	return m.minStack[len(m.minStack)-1]
}

func min(a, b int) int {
	if a > b {
		return b
	}
	return a
}

File: leetcode/python/_155/main.py
from typing import Optional

class MinStack:

    def __init__(self):
        self.stack = []
        self.minStack = []

    def push(self, val: int) -> None:
        self.stack.append(val)
        val = min(self.minStack[-1], val) if self.minStack else val
        self.minStack.append(val)

    def pop(self) -> None:
        self.stack.pop()
        self.minStack.pop()

    def top(self) -> int:
        return self.stack[-1]

    def getMin(self) -> int:
        return self.minStack[-1]

print(MinStack().push(-2))
print(MinStack().push(0))
print(MinStack().push(-3))
print(MinStack().getMin())
print(MinStack().pop())
print(MinStack().top())
print(MinStack().getMin())